Add production/consumption to topic list and topic overview

acass91 opened the pull request:
<!-- ignore-task-list-start -->
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)


<!-- ignore-task-list-end -->

- Added one column, multiple lines to the topic list for production and consumption
- Added production and consumption to the topic overview


**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)
<!-- ignore-task-list-start -->
- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation
<!-- ignore-task-list-end -->

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

![image](https://user-images.githubusercontent.com/6621993/197017019-f30a8351-6183-4eb8-8c13-8fb958ac8032.png)

Haarolean reviewed (changes requested):
PTAL at linter warnings
<img width="711" alt="image" src="https://user-images.githubusercontent.com/1494347/197027257-a639ed9c-0377-4f1f-b02e-dd51c2be9045.png">

Haarolean reviewed (changes requested):
1. If there's just one of the two fields present (either in or out) there's just one line and it looks ugly
<img width="552" alt="image" src="https://user-images.githubusercontent.com/1494347/197281012-35fcd3f9-a224-421c-9655-8b9108832fe9.png">
ugly as well with both fields:
<img width="419" alt="image" src="https://user-images.githubusercontent.com/1494347/197291403-d12c49cb-4fb5-4393-8cee-f67d519aa8c6.png">

2. please keep the topic row height fixed, let's make in/out font size smaller

@provectus/kafka-frontend please review the code

acass91 commented:
> 

That's largely subjective. We could be going back and forth for days without a specification for the UI. Is there an exact font and font size that you want? I used the defaults for that table.

Is there an example that doesn't use `<tr>` and `<td>` because I think that is what's causing the line.

Haarolean commented:
> > 
> 
> That's largely subjective. We could be going back and forth for days without a specification for the UI. Is there an exact font and font size that you want? I used the defaults for that table.
> 
> Is there an example that doesn't use `<tr>` and `<td>` because I think that is what's causing the line.

It might be. What's not subjective is that topic row height shouldn't be random and dynamic.

Let's wait for the design in this case. 
If you're participating in hacktoberfest, I can put an extra label to the PR if you want.

acass91 commented:
Agreed, the row height can be fixed. Hacktoberfest label would be awesome, thanks.

workshur commented on kafka-ui-react-app/src/components/Topics/List/TopicTable.tsx:
```diff
@@ -8,11 +8,57 @@ import { useSearchParams } from 'react-router-dom';
+          out: <BytesFormatted value={consumption} />
+        </td>
+      </tr>
+    </div>
```
Sorry, not sure I understand how this html construction should work in a table cell. And please let's avoid wrapping `tr` to `div`.

acass91 replied:
Can do. What's the best way to let the text span more than one line? That was specified in the issue.

acass91 replied:
Seems to work without using `tr' and `td`.

workshur reviewed (changes requested):
1. Please create custom Cell component for such logic - [Cell API](https://tanstack.com/table/v8/docs/api/core/cell)
2. This component should not contain `<tr><td>` - only value
More info about Column Defs https://tanstack.com/table/v8/docs/guide/column-defs#accessor-functions

acass91 commented on kafka-ui-react-app/src/components/Topics/List/TopicTable.tsx:
```diff
@@ -8,11 +8,57 @@ import { useSearchParams } from 'react-router-dom';
 import ActionsCell from './ActionsCell';
 import BatchActionsbar from './BatchActionsBar';
 
+function formatThroughput(row: Topic) {
```
Done

acass91 commented on kafka-ui-react-app/src/components/Topics/List/TopicTable.tsx:
```diff
@@ -85,6 +131,14 @@ const TopicTable: React.FC = () => {
       },
+      {
+        header: 'Throughput',
+        accessorFn: (row) => formatThroughput(row),
```
Done

workshur commented on kafka-ui-react-app/src/components/Topics/List/ThroughputCell.tsx:
```diff
@@ -0,0 +1,43 @@
+  const production = row.bytesInPerSec;
+  const consumption = row.bytesOutPerSec;
+  if (production === undefined && consumption === undefined) {
+    return <div>N/A</div>;
```
Can we use Fragment insted of div here?

workshur replied:
```suggestion
    return <>N/A</>;
```

workshur commented on kafka-ui-react-app/src/components/Topics/List/ThroughputCell.tsx:
```diff
@@ -0,0 +1,43 @@
+  }
+  if (production === undefined) {
+    return (
+      <div>
+        out: <BytesFormatted value={consumption} />
+      </div>
```
```suggestion
      <>
        out: <BytesFormatted value={consumption} />
      </>
```

workshur commented on kafka-ui-react-app/src/components/Topics/List/ThroughputCell.tsx:
```diff
@@ -0,0 +1,43 @@
+  }
+  if (consumption === undefined) {
+    return (
+      <div>
+        in: <BytesFormatted value={production} />
+      </div>
```
```suggestion
      <>
        in: <BytesFormatted value={production} />
      </>
```

workshur commented on kafka-ui-react-app/src/components/Topics/List/ThroughputCell.tsx:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  return (
+    <div>
+      <div>
+        in: <BytesFormatted value={production} />
+      </div>
+      <div>
+        out: <BytesFormatted value={consumption} />
+      </div>
+    </div>
```
```suggestion
    <>
      <div>
        in: <BytesFormatted value={production} />
      </div>
      <div>
        out: <BytesFormatted value={consumption} />
      </div>
    </>
```

workshur commented:
looks good in general, just a small nit to fix

Mgrdich commented on kafka-ui-react-app/src/components/Topics/List/ThroughputCell.tsx:
```diff
@@ -0,0 +1,43 @@
+import { Topic } from 'generated-sources';
+import BytesFormatted from 'components/common/BytesFormatted/BytesFormatted';
+
+function formatThroughput(row: Topic) {
```
why are we creating extra function reference in the memory and we only use it in a single place , i think we can move this logic to the component.

Haarolean reviewed (changes requested):
Hey @acass91, we've prepared a design solution [here](https://www.figma.com/file/ZkvysdRUmt9p2PDNU3a9WU/Kafka-UI?node-id=3175%3A12097&t=s4XHQGtMO5tWYspP-0). Please take a look 🙏🏼

acass91 commented:
> Hey @acass91, we've prepared a design solution [here](https://www.figma.com/file/ZkvysdRUmt9p2PDNU3a9WU/Kafka-UI?node-id=3175%3A12097&t=s4XHQGtMO5tWYspP-0). Please take a look 🙏🏼

Thanks. I added the SVGs for the arrows and changed the heading to 'Speed'. The metrics (bytesInPerSec, bytesOutPerSec) appear to have stopped working since I last ran this.

Haarolean commented:
@acass91 thanks. We'll get back to your PR once we do a release, currently we're in a code freeze stage. Thanks!

Mgrdich commented on kafka-ui-react-app/src/components/Topics/List/ThroughputCell.tsx:
```diff
@@ -0,0 +1,41 @@
+  }
+  if (production === undefined) {
+    return (
+      <>
```
do we need this `React` empty fragments  `<></>` ?

Mgrdich replied:
can we please check this as well.

David-DB88 commented:
@Haarolean i'll try

Mgrdich commented on kafka-ui-react-app/src/components/Topics/List/ThroughputCell.tsx:
```diff
@@ -0,0 +1,71 @@
+  }
+  if (production === undefined) {
+    return (
+      <Wrapper>
```
we can refactor this part in a tiny component. and use it in this conditions.

one question why are we checking `if (production === undefined) ` instead of `if(!production)`.

David-DB88 replied:
@ i just change conditions take a look please if its normal ?
